app/scripts/lib/rpc-method-middleware/handlers/watch-asset.js

```diff
@@ -37,14 +36,10 @@ async function watchAssetHandler(
 ) {
   try {
     const { options: asset, type } = req.params;
-    const handleWatchAssetResult = await handleWatchAssetRequest(asset, type);
-    await handleWatchAssetResult.result;
+    await handleWatchAssetRequest(asset, type);
     res.result = true;
     return end();
   } catch (error) {
-    if (error.message === 'User rejected to watch the asset.') {
-      return end(ethErrors.provider.userRejectedRequest());
```

Comment:
This is no longer needed, since that's the error we are defining in the FE component for rejection.